src/test/librust/wallet_zkeys_tests.cpp

```diff
@@ -116,7 +116,12 @@ BOOST_AUTO_TEST_CASE(StoreAndLoadSaplingZkeys) {
     BOOST_CHECK_EQUAL(wallet.GetSaplingScriptPubKeyMan()->mapSaplingZKeyMetadata[ivk2].nCreateTime, now);
 
     // Load a diversified address for the third key into the wallet
+    libzcash::SaplingPaymentAddress defaultAddr2 = sk2.DefaultAddress();
     auto dpa2 = sk2.ToXFVK().Address(diversifier).get().second;
+    while (dpa2 == defaultAddr2 && diversifier.begin()[0] < 255) {
+        diversifier.begin()[0] += 1;
```

Comment:
would be good to return the diversifier to 0 before enter in the loop. It could have been increased by the first loop.

Reply:
Sure :+1: Updated.